src/VisualStudio/Core/Next/Remote/ServiceHubRemoteHostClient.cs

```diff
@@ -200,5 +205,47 @@ private void OnRpcDisconnected(object sender, JsonRpcDisconnectedEventArgs e)
             // request service to HubClient timed out, more than we are willing to wait
             throw new TimeoutException("RequestServiceAsync timed out");
         }
+
+        private static int ReportDetailInfo(IFaultUtility faultUtility)
+        {
+            // 0 means send watson, otherwise, cancel watson
+            // we always send watson since dump itself can have valuable data
+            var exitCode = 0;
+
+            try
+            {
+                var logPath = Path.Combine(Path.GetTempPath(), "servicehub", "logs");
+                if (!Directory.Exists(logPath))
+                {
+                    return exitCode;
+                }
+
+                // attach all log files that are modified less than 1 day before.
+                var now = DateTime.UtcNow;
+                var oneDay = TimeSpan.FromDays(1);
+
+                foreach (var file in Directory.EnumerateFiles(logPath, "*.log"))
+                {
+                    var lastWrite = File.GetLastWriteTimeUtc(file);
```

Comment:
@IlyaBiryukov @AArnott is there a way for me to filter more on the logs? since all vs instances dump logs to the same folder and not clear any log ever created, just blindly adding all files in the log folder seems like a potential issue.